server/src/main/java/io/crate/interval/NumericalIntervalParser.java

```diff
@@ -97,8 +97,16 @@ private static Period roundToPrecision(int value,
                 return Period.seconds(value);
             }
         }
-        if (start == IntervalParser.Precision.SECOND && end == null) {
-            return buildSecondsWithMillisPeriod(value, millis);
+        if (start == IntervalParser.Precision.SECOND) {
+            if (end == null) {
+                return buildSecondsWithMillisPeriod(value, millis);
+            }
+            if (end == IntervalParser.Precision.MILLISECOND) {
+                return buildSecondsWithMillisPeriod(0, value * 1000 + millis);
+            }
+        }
```

Comment:
I think we can revert NumericalIntervalParser changes. 

If start is SECONDS, we should never truncate it, ie shouldn't do `buildSecondsWithMillisPeriod(0...`

Pretty much same logic as in another comment - millis it the smallest unit for now, we cannot start with it and we have nothing below to reset.

Reply:
Agree. After checking in Postgres, the seconds part is treated as one unit. The milliseconds are port of the seconds.